sympy/combinatorics/rewritingsystem.py

```diff
@@ -400,8 +488,24 @@ def reduce_using_automaton(self, word):
         '''
         # compute all_rules.
         all_rules = self.rules
-        self.compute_inverse_rules()
-        all_rules.update(self.inverse_rules)
+        inverse_rules = self.compute_inverse_rules(all_rules)
+        all_rules.update(inverse_rules)
```

Comment:
You are recomputing `all_rules` again. Weren't you going to make the right hand sides of rules part stored in each state as a `subst` attribute?

Reply:
I will be pushing that with these changes.